### PR TITLE
Create tech-spec.md

### PR DESCRIPTION
**Pull request by margielamadman**
This is our first pass at the tech spec. Tell us what you think.

### CODE REVIEW

**Comment by slively** [comment]
This is a good first step to lay out the tech stack and we want to elaborate by expanding on each section. I'm happy to spend time on Tuesday if you get stuck.

(btw you can put the image for the drawing in the repo and link to it from the readme)

React:
- what are the different pages
- what communicating does it do with the node server and when

Node:
- What are the different api endpoints and what are the contracts (raml, swagger, ...)

Postgres:
- What are the high level tables and their relationships

Kubernetes:
- How does Kubernetes decide which requests go to nginx and which requests go to node? (A common pattern is based on path, so /foo/* always goes to the foo service, /bar/* always goes to the bar service, etc...)
- The static Gatsby files will be inside the container with nginx, so the drawing is a little off.

File: tech-spec.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+* Volunteer form on each individual project
+  - Store info in local storage.
+
+### Nginx
+Serving the static Gatsby pages.
+
+### Node server
+* Express for routing/api.
+* Typescript is the language choice.
+* Send email using Gmail API.
```
Comment (oliverdain) [inline tech-spec.md]: I think it'll actually be the google cloud API. Sounds like the same thing but it isn't.

File: tech-spec.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+* Slack integreation when new projects are proposed.
+
+### Postgres DB
+To store our project info.
+
+### Gatsby
+Where our current static pages are made.
+
+### Kubernetes
+Hosting everything but Gatsby.
```
Comment (oliverdain) [inline tech-spec.md]: The Gatsby stuff is already hosted in our Kubernetes cluster. To clarify, Gatsby is a tool to generate static web pages from Markdown, etc. The *generated* pages are hosted in a container in our kubernetes cluster. That container holds nginx and the static assets generated by Gatsby. But the code generation part of gatsby isn't really hosted anywhere so this sentence is a little confusing.

**Comment by jasepellerin** [comment]
Made a couple changes to the spec, going to add the additional pieces from @slively's comment in a separate PR.

File: spec/tech-spec.md [hunk 3]
```diff
@@ -49,6 +49,17 @@ Serving the static Gatsby output.
...
+    id: UUID, PRIMARY KEY
+    project_name: VARCHAR(256), NOT NULL
+    org_url: VARCHAR(2083)
+    po_name: VARCHAR(128), NOT NULL
+    po_email: VARCHAR(256), NOT NULL
+    description: VARCHAR(512), NOT NULL
+    description_link: VARCHAR(2083)
+```
+
 ### Gatsby
```
Comment (russell-osu) [inline spec/tech-spec.md]: Should we add a field for tracking status?

File: spec/tech-spec.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+    Return project info.
+* `/api/project` (POST)
+
+    Add a new project to the database.
+* `/api/project/{id}` (PUT)
+
+    Update an existing project.
+* `/api/projects` (GET)
+
```
Comment (russell-osu) [inline spec/tech-spec.md]: Should we add an /api/project/{id} (DELETE) endpoint?

**Pull request merged**